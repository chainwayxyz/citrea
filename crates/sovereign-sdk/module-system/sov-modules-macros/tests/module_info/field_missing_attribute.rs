use sov_modules_api::{Context, Module, ModuleInfo, StateMap, WorkingSet};

#[derive(ModuleInfo)]
struct TestStruct<C: Context> {
    #[address]
    pub address: C::Address,

    test_state1: StateMap<u32, u32>,

    #[state]
    test_state2: StateMap<Vec<u8>, u64>,
}

impl<C: Context> Module for TestStruct<C> {
    type Context = C;

    type Config = ();

    type CallMessage = ();

    type Event = ();

    fn call(
        &self,
        _message: Self::CallMessage,
        _context: &Self::Context,
        _working_set: &mut WorkingSet<Self::Context>,
    ) -> Result<sov_modules_api::CallResponse, sov_modules_api::Error> {
        todo!()
    }
}

fn main() {}
