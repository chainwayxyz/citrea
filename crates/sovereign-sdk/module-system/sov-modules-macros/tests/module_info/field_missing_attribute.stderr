error: The field `test_state1` is missing an attribute: add `#[kernel_module]`, `#[module]`, `#[state]`, `#[address]`, or #[gas].
 --> tests/module_info/field_missing_attribute.rs:8:5
  |
8 |     test_state1: StateMap<u32, u32>,
  |     ^^^^^^^^^^^
